text/0003-templates.md

```diff
@@ -0,0 +1,393 @@
+
+## Detailed Explanation
+
+A "*Template*" is a normal `package.json` file with the `pnpm.template` field set.
+
+```json5
+{
+  "name": "@example/frontend-catalog",
+  "pnpm": {
+    "template": "catalog"
+  },
+  "dependencies": {
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "redux": "^4.2.0",
+    "react-redux": "^8.0.0"
+  }
+}
+```
+
+A package can then reference the template to populate different portions of its definition. Templates are available in different flavors that provide distinct features.
+
+- `catalog` — templates that share dependency specifiers
+- `toolkit` — templates that share `devDependencies`
+- `authoring` — templates that share `author`, `license`, `funding`, `repository`, `bugs`, `contributors`
+- `scripts` — templates that share scripts
+- `compatibility` — templates that can modify the package manifest of other dependencies through `pnpm.packageExtensions`. (Alternative to the compatibility DB.)
+- `patches` — templates that can modify the package manifest and contents of dependencies through fields such as `pnpm.packageExtensions`, `pnpm.overrides`, `pnpm.patchedDependencies`, `pnpm.peerDependencyRules`, etc.
+
+Templates must opt into being a template by defining `pnpm.template` and the flavor. This is a safety mechanism to:
+
+1. Intentionally prevent existing packages from being referred to for this purpose. Templates have different obligations and usage than standard packages that authors of templates should be aware of.
+2. Allow checks to be performed on the template before publishing and during consumption. Not all `package.json` fields may be valid on a template.
+
+### Authoring
+
+The below shows an example of a simple `authoring` template.
+
+```json5
+{
+  "name": "@example/authoring-template",
+  "version": "0.1.0",
+  "pnpm": {
+    "template": "authoring"
+  },
+  "author": "Example Organization <team@organization.example>",
+  "license": "MIT",
+  "repository": "git@organization.example/example/project"
+}
+```
+
+A package referencing the `@example/authoring-template` above will have the following on-disk and in-memory representations.
+
+**On-Disk**
+
+```json5
+{
+  "name": "@example/react-components",
+  "pnpm": {
+    "templates": {
+      "authoring": ["@example/authoring-template@0.1.0"],
+    }
+  }
+}
+```
+
+**In-Memory and Publish Time**
+
+```json5
+{
+  "name": "@example/react-components",
+  "author": "Example Organization <team@organization.example>",
+  "license": "MIT",
+  "repository": "git@organization.example/example/project"
+}
+```
+
+### Toolkit
+
+The `pnpm.templates.toolkit` flavor copies the `dependencies` block of the template and spreads it into the `devDependencies` of the referencing package.
```

Comment:
Why not copy `devDependencies` to `devDependencies` instead?

Reply:
@dominikg had the same concern and I think this is reasonable. @zkochan Should we simply copy `devDependencies` to `devDependencies`?